From: Ray Strode <rstrode@redhat.com>
Date: Fri, 4 Mar 2022 11:51:57 -0500
Subject: data: Disable wayland if there are multiple virtual gpus

Mutter doesn't seem to work with multiple guest gpus right now,
so disable that.

(cherry picked from commit 95a384a6464f4aca8cdb2a8cc03adc2d1e050e94)
---
 data/61-gdm.rules.in | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/data/61-gdm.rules.in b/data/61-gdm.rules.in
index d45f1c1..cc9f764 100644
--- a/data/61-gdm.rules.in
+++ b/data/61-gdm.rules.in
@@ -81,6 +81,13 @@ TEST!="/run/udev/gdm-machine-has-hardware-gpu", GOTO="gdm_virt_passthrough_check
 GOTO="gdm_disable_wayland"
 LABEL="gdm_virt_passthrough_check_end"
 
+# Disable wayland when there are multiple virtual gpus
+LABEL="gdm_virt_multi_gpu_check"
+TEST!="/run/udev/gdm-machine-has-hybrid-graphics", GOTO="gdm_virt_multi_gpu_check_end"
+TEST!="/run/udev/gdm-machine-has-virtual-gpu", GOTO="gdm_virt_multi_gpu_check_end"
+TEST=="/run/udev/gdm-machine-has-hardware-gpu", GOTO="gdm_virt_multi_gpu_check_end"
+LABEL="gdm_virt_multi_gpu_check_end"
+
 GOTO="gdm_end"
 
 LABEL="gdm_prefer_xorg"
